OpenSimRT/RealTime/experimental/RealTimeAnalysisExtended/include/RealTimeAnalysisExtended.h

```diff
@@ -0,0 +1,109 @@
+/**
+ * -----------------------------------------------------------------------------
+ * Copyright 2019-2021 OpenSimRT developers.
+ *
+ * This file is part of OpenSimRT.
+ *
+ * OpenSimRT is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * OpenSimRT is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * OpenSimRT. If not, see <https://www.gnu.org/licenses/>.
+ * -----------------------------------------------------------------------------
+ *
+ * @file RealTimeAnalysisExtended.h
+ *
+ * @brief This file contains a facade class that extends the RealTimeAnalysis
+ * class by including experimental features, such as GRFMPrediction and
+ * MarkerReconstruction.
+ *
+ * @author Filip Konstantinos <filip.k@ece.upatras.gr>
+ */
+#pragma once
+
+#include "GRFMPrediction.h"
+#include "MarkerReconstruction.h"
+#include "RealTimeAnalysis.h"
+#include "internal/RealTimeExports.h"
+#include <SimTKcommon/SmallMatrix.h>
+
+namespace OpenSimRT {
+
+/**
+ * Phase detector update function that uses external measurements.
+ */
+typedef std::function<void()> ExternalPhaseDetectorUpdateFunction;
+
+/**
+ * Phase detector update function that uses internal estimations from the
+ * musculoskeletal analysis.
+ */
+typedef std::function<void(const double& t, const SimTK::Vector& q,
+                           const SimTK::Vector& qd, const SimTK::Vector& qdd)>
+        InternalPhaseDetectorUpdateFunction;
+
+/**
+ * @brief Extends the facade RealTimeAnalysis class to include the experimental
+ * features. Provides a convinient interface for performing RT musculoskeletal
+ * analysis. It creates one thread for the data acquisition, marker
+ * reconstruction IK and filtering, and one processing thread for the rest the
+ * analysis (GRF&M prediction, ID, SO and JR).
+ */
+class RealTime_API RealTimeAnalysisExtended : public RealTimeAnalysis {
```

Comment:
Is it necessary to extend from RealTimeAnalysis? Maybe we can decouple them.

Reply:
I did it for code reuse. Practically only the added modules and their parameters differ.